packages/cardano-services/src/StakePoolSearch/StakePoolSearchHttpServer.ts

```diff
@@ -40,29 +34,30 @@ export class StakePoolSearchHttpServer extends HttpServer {
   ) {
     const router = express.Router();
     router.use(bodyParser.json({ limit: config.bodyParser?.limit || '500kB', type: 'application/json' }));
-    router.post('/search', async (req, res) => {
-      if (req.header('Content-Type') !== 'application/json') {
-        res.statusCode = 400;
-        return res.send('Must use application/json Content-Type header');
-      }
-      logger.debug('/search', { ip: req.ip });
-      let body: Error['message'] | StakePoolSearchResponse;
-      try {
-        const response = await stakePoolSearchProvider.queryStakePools(req.body);
-        body = { stakePools: response };
-      } catch (error) {
-        // TODO: catch db errors
-        res.statusCode = 500;
-        body = JSON.stringify(
-          Array.isArray(error) ? error.map<ErrorObject>((e) => serializeError(e)) : serializeError(error)
-        );
-        logger.error(body);
-      }
-      res.send(body);
-    });
+    router.post(
+      '/search',
+      providerHandler<StakePoolQueryOptions>(async ([stakePoolOptions], req, res) => {
+        if (req.header('Content-Type') !== 'application/json') {
```

Comment:
This is redundant, it would have empty body and fail [here](https://github.com/input-output-hk/cardano-js-sdk/pull/215/files#diff-982516a6fc5397db8d37bc455abffec843ec2c9c90f403be07f45ece2dffb110R12)